app/assets/javascripts/services/topology_service.js

```diff
@@ -216,26 +217,20 @@ ManageIQ.angular.app.service('topologyService', function() {
   // this injects some common code in the controller - temporary pending a proper merge
   this.mixinSearch = function($scope) {
     var topologyService = this;
-
+      // NOTE: listener on search
+    ManageIQ.angular.rxSubject.subscribe(function(event) {
+      if (event.service === 'topologyService') {
+        var query = $('input#search_topology')[0].value;
+        var svg = topologyService.getSVG($scope.d3);
```

Comment:
svg is only used in the `searchNode` case .. probably shouldn't call this outside that if